### ISSUE TITLE
[FEATURE] QoL epubs

### ISSUE THREAD

**Issue by hollisticated-horse** [stumpapp/stump#378]
**Is your feature request related to a problem? Please describe.**
A variety of things that would improve reading EPUBs. 

**Describe the solution you'd like**
##### WIP Proposal:

- adjustable line-heights
- word spacing, by having some presets ?
- margins around the whole book (top, bottom, left and right synced)
- adjustable text-justify → left, right, center and justified. 
- have a progress bar instead of page numbers
   - interactive, be able to see small icons indicating highlights & bookmarks (#115 )

**Describe alternatives you've considered**
n.a.

**Additional context**
Based of 
- KOReader 
- BookFusion
- and other reading apps.

**Comment by hollisticated-horse** [comment]
I'd be happy to make some design drafts in a penpot we could share @aaronleopold

**Comment by aaronleopold** [comment]
> have a progress bar instead of page numbers
interactive, be able to see small icons indicating highlights & bookmarks

I'm personally a HUGE fan of Yomu's progress bar, if you have seen it.

> I'd be happy to try to make some design drafts in a penpot we could share

Absolutely! Any design help is extremely appreciated 🙂

**Comment by hollisticated-horse** [comment]
> I'm personally a HUGE fan of Yomu's progress bar, if you have seen it.

Oooh, I have not. It's mac/ios specific, but it would be really cool if you could find screenshots. :D
From the screenshot in the main page of the project, it does have a fairly minimalistic look. What is good about it ? 

> Absolutely! Any design help is extremely appreciated 🙂

I'll make a penpot project later, if you'd like ? I can send you an invite by email. 
I'm a bit busy, but in a few weeks I should have more time. :D

**Comment by aaronleopold** [comment]
> From the screenshot in the main page of the project, it does have a fairly minimalistic look. What is good about it ?

Yeah it's very minimalist, I'm just preferential to that kind of design in general. It chunks each chapter into blocks, then shows your relative position in the block for the current chapter. This is very difficult to do with EPUB.js, though.

> I find that the bookfusion one is decent, similar to foliate (linux, flathub)

I haven't given bookfusion a proper look, a quick glance at the [preview](https://www.bookfusion.com/reading) looks good too

> I'll make a penpot project later, if you'd like ? I can send you an invite by email.
> I'm a bit busy, but in a few weeks I should have more time. :D

No rush or expectations on my end. I haven't used penpot before, but [email](mailto:aaronleopold1221@gmail.com) works

**Comment by hollisticated-horse** [comment]
Here a first proposition: 

https://design.penpot.app/#/view/c0ee57fd-603e-804a-8004-c6d33343fd51?page-id=c0ee57fd-603e-804a-8004-c6d33343fd52&section=interactions&index=0

**Comment by aaronleopold** [comment]
> https://design.penpot.app/#/view/c0ee57fd-603e-804a-8004-c6d33343fd51?page-id=c0ee57fd-603e-804a-8004-c6d33343fd52&section=interactions&index=0

I don't seem to have access. I thought maybe I would need an account to access, but still get the same error after creating one and trying the link:

> This page might not exist or you don’t have permissions to access to it.

**Comment by aaronleopold** [comment]
I just took a look. I think it's good!

> I'm thinking that the sections in the progress bar should reflect the toc and page counts

Yeah, that's how Yomu does it too. Each chapter (toc entry) is a block, and the fill is based on your current page in the current chapter. The caveat would be offhand I don't know how simple calculating that would be, mostly because to generate accurate sizing of previous chapters would mean we know how many pages are in each which is a lot of computation that would be reactive to any changes in the viewport size

**Comment by hollisticated-horse** [comment]
Just tested 0.11. The improvements are really impressive !!
I updated the issued to reflect what has been done.
